Code/Framework/AzCore/AzCore/Name/Name.cpp

```diff
@@ -19,7 +19,10 @@
 
 namespace AZ
 {
-    AZStd::thread::id Name::s_staticNameListThread{};
+    static const AZStd::thread::id InvalidThreadId;
```

Comment:
This constant is not needed.
The code just can compare against `AZStd::thread::id{}`

Reply:
Ok, then can you move it to the SetNameLiteral function and make it constexpr?

Reply:
I've removed the variable since `s_staticNameListThread` was removed too.